src/Corporation/Actions.ts

```diff
@@ -198,125 +198,120 @@ export function acceptInvestmentOffer(corporation: Corporation): void {
   corporation.investorShares += investShares;
 }
 
-export function sellMaterial(material: Material, amount: string, price: string): void {
-  if (price === "") price = "0";
-  if (amount === "") amount = "0";
-  let cost = price.replace(/\s+/g, "");
-  cost = cost.replace(/[^-()\d/*+.MPe]/g, ""); //Sanitize cost
-  let temp = cost.replace(/MP/, "1.234e5");
+export function convertPriceString(price: string): string | number {
+  // Replace whitespace characters.
+  let sanitizedPrice = price.replace(/\s+/g, "");
```

Comment:
This is completely unneeded; because whitespace characters are not part of the set you define below, they will automatically be squashed.

That said, I'm not sure they *should* be. If someone enters "123  456", should that be interpreted as "123456"? Bear in mind that `parseFloat` functions fine with leading and/or trailing whitespace.